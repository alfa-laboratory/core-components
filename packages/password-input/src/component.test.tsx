import React, { useState, FC } from 'react';
import { render, fireEvent } from '@testing-library/react';

import { PasswordInput, PasswordInputProps } from '.';

const ControlledPasswordInput: FC<PasswordInputProps> = props => {
    const [passwordVisible, setPasswordVisible] = useState(false);

    const handlePasswordVisibleChange = (visible: boolean) => {
        setPasswordVisible(visible);
    };

    return (
        <PasswordInput
            {...props}
            passwordVisible={passwordVisible}
            onPasswordVisibleChange={handlePasswordVisibleChange}
        />
    );
};

const isPasswordHidden = (input: HTMLInputElement) => input.getAttribute('type') === 'password';
const isPasswordVisible = (input: HTMLInputElement) => input.getAttribute('type') === 'text';

describe('PasswordInput', () => {
    const dataTestId = 'test-id';

    describe('snapshots tests', () => {
        it('should match snapshot', () => {
            expect(render(<PasswordInput />).baseElement).toMatchSnapshot();
        });
    });

    describe('interactions tests', () => {
        it('should switch password visibility (uncontrolled)', () => {
            const { baseElement, getByTestId } = render(<PasswordInput dataTestId={dataTestId} />);

            const button = baseElement.querySelector('button') as HTMLButtonElement;
            const input = getByTestId(dataTestId) as HTMLInputElement;

            expect(isPasswordHidden(input)).toBe(true);

            fireEvent.click(button);

            expect(isPasswordVisible(input)).toBe(true);

            fireEvent.click(button);

            expect(isPasswordHidden(input)).toBe(true);
        });

        it('should switch password visibility (controlled)', () => {
            const { baseElement, getByTestId } = render(
                <ControlledPasswordInput dataTestId={dataTestId} />,
            );

            const button = baseElement.querySelector('button') as HTMLButtonElement;
            const input = getByTestId(dataTestId) as HTMLInputElement;

            expect(isPasswordHidden(input)).toBe(true);

            fireEvent.click(button);

            expect(isPasswordVisible(input)).toBe(true);

            fireEvent.click(button);

            expect(isPasswordHidden(input)).toBe(true);
        });
    });
});
