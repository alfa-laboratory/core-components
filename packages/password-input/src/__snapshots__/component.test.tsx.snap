// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordInput snapshots tests should match snapshot 1`] = `
<body>
  <div>
    <div
      class="component component s s hasRightAddons"
    >
      <div
        class="inner inner"
      >
        <div
          class="inputWrapper"
        >
          <div
            class="input"
          >
            <input
              class="input input input"
              type="password"
              value=""
            />
          </div>
        </div>
        <div
          class="addons rightAddons addons"
        >
          <button
            class="component ghost s component ghost iconOnly eye secondary"
            type="button"
          >
            <span
              class="addons"
            >
              <span
                class="iconWrapper s"
              >
                <svg
                  class="icon"
                  fill="currentColor"
                  focusable="false"
                  height="24"
                  role="img"
                  viewBox="0 0 24 24"
                  width="24"
                >
                  <path
                    d="M20.944 14.702a.197.197 0 00.293-.014C21.886 13.89 22.564 12.93 23 12c-1.914-4.089-6.19-7.01-11-7.01-.164 0-.326.003-.489.01-.17.007-.246.022-.125.143l9.558 9.559z"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M5.5 2L22 18.5 20.5 20l-2.44-2.527C16.28 18.561 14.21 19 12 19c-4.81 0-9.086-2.911-11-7 1.155-2.468 3.252-4.633 5.69-5.896L4 3.5 5.5 2zm8.177 11.09l1.571 1.573a4.2 4.2 0 11-5.91-5.911l1.571 1.571a2 2 0 102.767 2.767z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
